source/ShipyardPanel.cpp

```diff
@@ -307,6 +307,15 @@ void ShipyardPanel::Sell(bool toStorage)
 	static const int MAX_LIST = 20;
 
 	int count = playerShips.size();
+	if(count == 1 && player.UuidLocked(playerShip->UUID()))
```

Comment:
A similar dialog could be shown if we have multiple ships selected and they're all locked.